doc/nrf/working_with_nrf/nrf70/developing/powersave.rst

```diff
@@ -219,15 +224,45 @@ Following are the two key parameters of Target Wake Time:
 
 * **TWT Wake Duration**
 
-  * The minimum amount of time that the TWT requesting device needs to be active to complete the frame exchanges during the TWT wake interval.
+  * The amount of time that the TWT requesting device needs to be active to complete the frame exchanges during the TWT wake interval.
   * Valid range for duration is 1 ms to 256 ms.
+    Lower value results in more power saving at the cost of potential loss of data.
+    It is recommended not to go lower than 8 ms.
+    The application can choose lower value than this at the cost of losing application data in the network.
   * The application must choose the right duration based on the traffic pattern.
+  * Applications must take appropriate action if uplink or downlink traffic is more than anticipated.
+    It could terminate the ongoing TWT session and negotiate a fresh session based on new requirements.
+
+  * Downlink traffic:
+
+    a. Predictability is the key for choosing the correct wake duration.
+    #. Wake duration must be sufficient for the AP to schedule all the incoming traffic to devices.
+       AP needs to contend the channel for scheduling frames and frames will be dropped if duration is aggressive in busy channels.
+    #. AP will drop the device data if it cannot finish all transmission in the wake duration.
+       AP may buffer traffic till the next interval if sleep duration is in the order of 100 milliseconds.
+       It will not buffer the device data if sleep duration is in the order of minutes and data will be lost.
+    #. Devices are allowed to sleep after the wake duration and there is no mechanism to extend the wake duration based on downlink traffic.
+       Device must be active during the wake duration even if there is no downlink traffic.
+
+  * Uplink traffic:
+
+    a. Trigger Enabled - When operating in Trigger Enabled mode, the nRF70 Series device:
+
+       * schedules uplink traffic as a response to trigger frames from the AP.
+       * expects the AP to schedule trigger frames in the wake duration.
+       * discards all pending frames after wake time expiry before entering the sleep state.
+
+    #. Non-trigger Enabled - When operating in Non-trigger Enabled mode, the nRF70 Series device:
+
+       * schedules uplink traffic using the legacy channel contention.
+       * tries to schedule all uplink traffic in the wake duration and discard pending frames.
+       * discards all pending frames after wake time expiry before entering the sleep state.
 
 * **TWT Wake Interval**
 
   * Interval between successive TWT wake periods.
   * Valid range for duration is one ms to a few days.
-  * The application must choose the right interval based on the traffic in receive direction.
+  * Application must choose the right interval based on the traffic in receive direction.
```

Comment:
@udaynordic Why specifically in the receive direction ? Surely it is equally applicable in both directions ? And should we say "... based on the **expected** traffic".

Reply:
Agree Mike. Its applicable for both directions.